src/NosCore.GameObject/Character.cs

```diff
@@ -678,36 +681,36 @@ public void Buy(Shop shop, short slot, short amount)
             if (inv?.Count > 0)
             {
                 inv.ForEach(it => it.CharacterId = CharacterId);
-                var packet = shop.Session?.Character.BuyFrom(item, amount, slotChar);
+                var packet = await (shop.Session == null ? Task.FromResult((NInvPacket?)null) : shop.Session.Character.BuyFrom(item, amount, slotChar));
                 if (packet != null)
                 {
-                    SendPacket(packet);
+                    await SendPacket(packet);
                 }
 
-                SendPackets(
+                await SendPackets(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5262615)